File: clients/python/src/cien_agent_sdk/public/user_company_permissions.py
from __future__ import annotations

from typing import Any, Literal

from ..base import EndpointGroup


class PublicUserCompanyPermissionsAPI(EndpointGroup):
    """/api/user-company-permissions endpoints."""

    def list_companies_for_user(
        self,
        *,
        email: str,
        role: Literal["view", "manage", "owner", "any"] = "any",
    ) -> list[dict[str, Any]]:
        """List companies a user can access, optionally filtered by role."""
        return self._get(f"/api/user-company-permissions/user/{email}/companies", params={"role": role})

    def list_users_for_company(
        self,
        *,
        coid: str,
        role: Literal["manage", "owner", "any"] = "any",
    ) -> list[dict[str, Any]]:
        """List users with access to a company, optionally filtered by role."""
        return self._get(f"/api/user-company-permissions/company/{coid}/users", params={"role": role})

    def set(
        self,
        *,
        email: str,
        coid: str,
        permission_role: Literal["view", "manage", "owner"],
    ) -> dict[str, Any]:
        """Set a user's permission role for a company."""
        return self._post(
            "/api/user-company-permissions/set",
            json={"email": email, "coid": coid, "permission_role": permission_role},
        )

    def remove(self, *, email: str, coid: str) -> dict[str, Any]:
        """Remove any explicit company permission for a user."""
        return self._post(
            "/api/user-company-permissions/remove",
            json={"email": email, "coid": coid},
        )

File: clients/node/src/public/user-company-permissions.ts
import { EndpointGroup } from '../base.js'
import type { CompanyUserRoleFilter, PermissionRole, UserCompanyRoleFilter } from '../types.js'

export class PublicUserCompanyPermissionsAPI extends EndpointGroup {
  listCompaniesForUser(params: { email: string; role?: UserCompanyRoleFilter }) {
    return this.requestGet<Array<Record<string, unknown>>>(
      `/api/user-company-permissions/user/${params.email}/companies`,
      { params: { role: params.role ?? 'any' } }
    )
  }

  listUsersForCompany(params: { coid: string; role?: CompanyUserRoleFilter }) {
    return this.requestGet<Array<Record<string, unknown>>>(
      `/api/user-company-permissions/company/${params.coid}/users`,
      { params: { role: params.role ?? 'any' } }
    )
  }

  getCurrentUserPermissionForCompany(params: { coid: string }) {
    return this.requestGet<Record<string, unknown>>(
      `/api/user-company-permissions/company/${params.coid}/me`
    )
  }

  set(payload: { email: string; coid: string; permission_role: PermissionRole }) {
    return this.requestPost<Record<string, unknown>>('/api/user-company-permissions/set', {
      json: payload,
    })
  }

  remove(payload: { email: string; coid: string }) {
    return this.requestPost<Record<string, unknown>>('/api/user-company-permissions/remove', {
      json: payload,
    })
  }
}
